scorch panics on arm (raspberry pi)

## uname -a
`
Linux raspberrypi 4.19.42-v7+ #1219 SMP Tue May 14 21:20:58 BST 2019 armv7l GNU/Linux
`

```go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x4 pc=0x12724]

goroutine 1 [running]:
github.com/blevesearch/bleve/index/scorch.(*Scorch).openBolt(0x2af8000, 0xe0f1b8, 0x2af8000)
        /Users/darren/Sources/go/pkg/mod/github.com/blevesearch/bleve@v0.7.1-0.20190510153952-e0de6a8c9c5f/index/scorch/scorch.go:218 +0x12c
github.com/blevesearch/bleve/index/scorch.(*Scorch).Open(0x2af8000, 0x6, 0x280ac80)
        /Users/darren/Sources/go/pkg/mod/github.com/blevesearch/bleve@v0.7.1-0.20190510153952-e0de6a8c9c5f/index/scorch/scorch.go:160 +0x1c
github.com/blevesearch/bleve.openIndexUsing(0x28285a0, 0xb, 0x0, 0x2aba7d0, 0x0, 0x0)
        /Users/darren/Sources/go/pkg/mod/github.com/blevesearch/bleve@v0.7.1-0.20190510153952-e0de6a8c9c5f/index_impl.go:173 +0x434
github.com/blevesearch/bleve.Open(...)
        /Users/darren/Sources/go/pkg/mod/github.com/blevesearch/bleve@v0.7.1-0.20190510153952-e0de6a8c9c5f/index.go:286
```

Due to  https://github.com/golang/go/issues/599

--- a/index/scorch/scorch.go
+++ b/index/scorch/scorch.go
@@ -41,12 +41,14 @@ const Version uint8 = 2
 var ErrClosed = fmt.Errorf("scorch closed")
 
 type Scorch struct {
+	nextSegmentID uint64
+	stats         Stats
+	iStats        internalStats
+
 	readOnly      bool
 	version       uint8
 	config        map[string]interface{}
 	analysisQueue *index.AnalysisQueue
-	stats         Stats
-	nextSegmentID uint64
 	path          string
 
 	unsafeBatch bool
@@ -72,8 +74,6 @@ type Scorch struct {
 
 	onEvent      func(event Event)
 	onAsyncError func(err error)
-
-	iStats internalStats
 
 	pauseLock sync.RWMutex
 

--- a/index/scorch/snapshot_segment.go
+++ b/index/scorch/snapshot_segment.go
@@ -183,9 +183,9 @@ func (cfd *cachedFieldDocs) prepareField(field string, ss *SegmentSnapshot) {
 }
 
 type cachedDocs struct {
+	size  uint64
 	m     sync.Mutex                  // As the cache is asynchronously prepared, need a lock
 	cache map[string]*cachedFieldDocs // Keyed by field
-	size  uint64
 }
 
 func (c *cachedDocs) prepareFields(wantedFields []string, ss *SegmentSnapshot) error {

